testsuite/openmodelica/cruntime/optimization/basic/Makefile

```diff
@@ -58,12 +56,13 @@ TFCtestFlag.mos \
 TT.mos \
 
 
-# test that currently fail. Move up when fixed. 
+# test that currently fail. Move up when fixed.
 # Run make testfailing
 FAILINGTESTFILES = \
 TT2.mos \
 TT3.mos \
-
+testAlgLoop5.mos \
```

Comment:
I will not spend any time on the optimization functionality. I did waste time on it before, and I can just tell it is implemented in a horrible way. Because I am not aware of anyone using it and also no one maintaining it, I think we should be pragmatic and ignoring it for now.

But I can wait merging it if someone volunteers to investigate it.

Reply:
> We should invest at least some time to investigate why the tests are failing now. But I would also guess, that these are just to fragile.
> If we move them to failing they will stay there for the next few years...

This is true. However, we have to strike a balance. As I understand, this PR fixes a lot of issues in the simulation of models, that are affecting lots of users, and the coverage of MSL, which is one of our strategic goals. I am willing to pay some price on advanced features which are less used and get somehow badly affected.

Of course we should open a new issue regarding that issue, but I don't think we should delay the merge because of if.

My 2 cts.

Reply:
Note:
For testAlgLoop5 we have
`optimize(testAlgLoop5, tolerance = 1e-12,simflags="-lv LOG_IPOPT_ERROR -optimizerNP 1 -ipopt_init CONST -iit 0.2 -iim=none -iif=ReferenceFiles/testAlgLoop5_ref.mat");`

with `iim=none`, so changes of the intial equation should have no inpact.  
What makes sense is that `iif` get broken .... 

Maybe related to https://trac.openmodelica.org/OpenModelica/ticket/5195

Reply:
The optimization is affected by `removeEqualFunctionCalls`.

Reply:
I pasted my analysis here:
https://trac.openmodelica.org/OpenModelica/ticket/5458#comment:11

The effect is that removeEqualFunctionCalls has an effect of the loop/tearing in the model, which independently from `optimization`-appi and has an effect of `simulation`-appi, too